src/l1_fetcher.rs

```diff
@@ -34,6 +36,49 @@ pub enum L1FetchError {
     GetTx,
 }
 
+#[derive(Default)]
+struct L1Metrics {
+    // Metrics variables.
+    l1_blocks_processed: u64,
+    l2_blocks_processed: u64,
+    latest_l1_block_nbr: u64,
+    latest_l2_block_nbr: u64,
+
+    first_l1_block: Option<u64>,
+    last_l1_block: Option<u64>,
+}
+
+impl L1Metrics {
+    fn print(&mut self) {
+        if self.first_l1_block.is_none()
+            || self.latest_l1_block_nbr == 0
+            || self.latest_l2_block_nbr == 0
+        {
+            return;
+        }
+
+        let first_l1_block = self.first_l1_block.unwrap();
+        let progress = match self.last_l1_block {
+            Some(last_l1_block) => {
+                let total = last_l1_block - first_l1_block;
+                let cur = self.latest_l1_block_nbr - first_l1_block;
+                let perc = (cur * 100) / total;
+                format!("{perc}%")
+            }
+            None => "∞".to_string(),
```

Comment:
Currently this won't print any percentages if we don't supply an `end_block`, right? Could we use the latest (currently polled) L1 block number that exists in that case?

Reply:
> this won't print any percentages if we don't supply an end_block, right?

If `disable_polling == false`, the execution continues forever and then this won't print any percentages.

> Could we use the latest (currently polled) L1 block number that exists in that case?

With extremely long stretch, yeah, maybe we could, but it would be very far from correct.

I struggled quite a bit to provide any useful metrics from L1 blocks because of following behaviour:
- We don't literally process L1 blocks. We filter event logs with certain criteria and each filter batch spans over multiple L1 blocks.
- We see events from L1 blocks here & there. Similar story with transactions.

Because of that, I don't really consider the L1 block number _too_ useful, but that being said, the information is already there (CUR L1 & L2 block numbers). I don't think it's useful to have it again either.

One alternative could be to display just "PROCESSING", instead of "PROGRESS [ n %]" when polling is enabled?

Reply:
Ok. I changed it so that `last_l1_block` is now always set. When polling goes past it, the percentages should cap at 100%.